### PR TITLE
Avoid str methods on path

### PR DESCRIPTION
**Pull request by olliethomas**
try simpler gen df func

### CODE REVIEW

File: conduit/data/datasets/audio/ecoacoustics.py [hunk 1]
```diff
@@ -230,13 +230,12 @@ def _extract_metadata(self) -> None:
...
             paths.extend(self.base_dir.glob(f"**/*{ext}"))
-            str_paths = pd.Series([str(path.relative_to(self.base_dir)) for path in paths])
-
-            df = str_paths.str.rpartition(
-                "\\",
+            return pd.DataFrame.from_dict(
+                {
+                    "filePath": [path.parent for path in paths],
+                    "fileName": [path.name for path in paths],
+                }
```
Comment (olliethomas) [inline conduit/data/datasets/audio/ecoacoustics.py:237]: this isn't guaranteed to be in the same order is it....

File: conduit/data/datasets/audio/ecoacoustics.py [hunk 2]
```diff
@@ -230,13 +230,15 @@ def _extract_metadata(self) -> None:
         def gen_files_df(ext: Extension) -> pd.DataFrame:
             paths: list[Path] = []
             paths.extend(self.base_dir.glob(f"**/*{ext}"))
```
Comment (thomkeh) [inline conduit/data/datasets/audio/ecoacoustics.py:232]: what the hell is this code?

Comment (olliethomas) [inline conduit/data/datasets/audio/ecoacoustics.py:232]:
```suggestion
```

File: conduit/data/datasets/audio/ecoacoustics.py [hunk 3]
```diff
@@ -230,13 +230,15 @@ def _extract_metadata(self) -> None:
...
-
-            df = str_paths.str.rpartition(
-                "\\",
+            return pd.DataFrame.from_dict(
+                [
+                    {
+                        "filePath": str(path.relative_to(self.base_dir)),
+                        "fileName": path.name,
+                    }
+                    for path in paths
```
Comment (olliethomas) [inline conduit/data/datasets/audio/ecoacoustics.py:239]:
```suggestion
                    for path in self.base_dir.glob(f"**/*{ext}")
```

**Review by MylesBartlett — APPROVED** [review]

**Comment by thomkeh** [comment]
can this be merged?

**Comment by olliethomas** [comment]
> can this be merged?

not yet. Still debugging the preprocessing

**Pull request merged**